src/charm.py

```diff
@@ -35,83 +34,78 @@ def __init__(self, *args: typing.Any):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.certificates = certificates_observer.CertificatesObserver(self)
```

Comment:
I just have a question , is the certificate here used for communication between wazuh server and wazuh client? or is it used for external access to wazuh?

Reply:
The former. Wazuh will only be accessible by the agents. It also has an HTTP API but we won't be exposing it to the outside.